docs/src/InvariantTheory/it.md

```diff
@@ -12,38 +12,35 @@ Pages = ["it.md"]
 
 # Introduction
 
-Our basic setting in invariant theory consists of a group $G$, a field $K$, 
-a vector space $V$ over $K$ of finite dimension $n,$  and a representation $\rho: G \to \text{GL}(V)$
-of $G$ on $V$. We write $V^\ast$ for the dual vector space of $V$ and suppose that a fixed set of
-cooordinates $x = \{x_1, \dots, x_n\}\in V^*$ is chosen.
+The basic setting in this chapter consists of a group $G$, a field $K$, a vector space
+$V$ over $K$ of finite dimension $n,$  and a representation $\rho: G \to \text{GL}(V)$
+of $G$ on $V$. Furthermore, $V^\ast$ denotes the dual vector space of $V$ and 
+$x = \{x_1, \dots, x_n\}\subset V^*$ denotes a fixed set of cooordinates.
 
-
-The action of $G$ on $V$ induces an action of  $G$ on $V^\ast$: Set
+The action of $G$ on $V$ defines an action of  $G$ on $V^\ast$,
 
 $(\pi f)(v)=f(\rho(\pi^{-1}) v).$
 
-This, in turn, defines an action of $G$ on the graded symmetric algebra
+This extends naturally to an action of $G$ on the graded symmetric algebra
 
-$K[x] = K[x_1, \dots, x_n] \cong  K[V]=S(V^*)=\bigoplus_{d\geq 0} S^d V^*.$
+$K[x] = K[x_1, \dots, x_n] \cong K[V]:=S(V^*)=\bigoplus_{d\geq 0} S^d V^*.$
 
 The *invariants* of $G$ are the fixed points of this action, its *ring of invariants* is the graded subalgebra
 
-$K[x]^G\cong K[V]^G=\{f\in K[V] \mid \pi f=f {\text { for any }} \pi\in G\}\subset K[V].$
+$K[x]^G = K[x_1, \dots, x_n]^G\cong  K[V]^G=\{f\in K[V] \mid \pi f=f {\text { for any }} \pi\in G\}\subset K[V].$
```

Comment:
The notation `K[x]^G` to me always indicates a *right* action, i.e. a map `SET x GROUP -> SET`. That would suggest writing `\pi ^ f = f`. This would be consistent with our plan to do everything in Oscar based on right actions, like it is done in GAP and in MAGMA. Indeed, MAGMA even does this in invariant theory, see:
- http://magma.maths.usyd.edu.au/magma/handbook/text/1323 for their definition of `K[V]^g`
- http://magma.maths.usyd.edu.au/magma/handbook/text/1327 for their matrix action on polynomials: for a polynomial `f`, a matrix `g` and a coordinate vector `x`, they define `f^g(x) := f(x.g)`, which is indeed a right action. Translated to the notation used above:  `f^\pi(v) = f(v \rho(\pi))`

This is in contrast to the action described above in line 22/23, which is a left action (and has to invert or transpose the acting matrix to achieve that).

Reply:
Actually, the action Magma describes is a left action after all... Now I am confused what they do, I will check with actual code.

Reply:
For what it's worth, I think finvar.lib in Singular does right actions. So the matrix M =
( a b )
( c d ) 
acts on K[x, y] by mapping x to a*x + b*y and y to c*x + d*y. At least I would call this a right action because I would identify x with the vector (1 0) and then (1 0)*M is (a b).

Reply:
The hallmark of a right action is it must satisfy `(f^a)^b = f^{a*b}`  while for a left action it is `a*(b*f) = (a*b)*f` (or, if one uses the notation for right actions to write down a left action, one gets `(f^a)^b = f^{b*a}`. So, just take two matrices a,b that do not commute and compare (f^a)^b` and `f^{ab}`.

So that's what one should check for the Singular computations.


As to Magma, I verified they do implement a right action on polynomials via the above rules:
```
> K := QuadraticField(2);
> Aq := [ x / K.1 : x in [1, 1, -1, 1]];
> Bq := [ x * One(K) : x in [0, 1, 1, 0]];
> G := MatrixGroup<2, K | Aq, Bq>;
> P<x, y> := PolynomialRing(K, 2);
> f := x^2 + x * y + y^2;

# equal:
> f^(G.1*G.2);
3/2*x^2 + 1/2*y^2
> (f^G.1)^G.2;
3/2*x^2 + 1/2*y^2

# different
> (f^G.2)^G.1;
1/2*x^2 + 3/2*y^2
```

Reply:
It's not quite as easy to reproduce this in Singular / finvar.lib, because the action of a matrix on a polynomial is just not implemented. Doing the action by hand as in `evaluate_reynolds()` for example the results are exactly as in Magma.